product_supplierinfo_for_customer/views/product_views.xml

```diff
@@ -0,0 +1,136 @@
+<?xml version="1.0" encoding="UTF-8" ?>
+<!-- Copyright 2015 AvanzOSC
+     Copyright 2015-18 Tecnativa
+     Copyright 2017-19 ForgeFlow
+     Copyright 2019 Tecnativa - Pedro M. Baeza
+     License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl). -->
+<odoo>
+    <record id="product_customerinfo_form_view" model="ir.ui.view">
+        <field name="name">product.customerinfo.form.view</field>
+        <field name="model">product.customerinfo</field>
+        <field name="arch" type="xml">
+            <form string="Customer Information">
+                <group>
+                    <group string="Customer">
+                        <field
+                            name="product_id"
+                            groups="product.group_product_variant,base.group_multi_company"
+                            domain="[('product_tmpl_id', '=', product_tmpl_id)]"
+                            options="{'no_create_edit': True}"
+                        />
+                        <field
+                            name="name"
+                            string="Customer"
+                            context="{'res_partner_search_mode': 'customer'}"
+                        />
+                        <field name="product_name" string="Customer Product Name" />
+                        <field name="product_code" string="Customer Product Code" />
+                        <field
+                            name="product_id"
+                            domain="[('product_tmpl_id', '=', product_tmpl_id)]"
+                            groups="product.group_product_variant,base.group_multi_company"
+                            invisible="1"
```

Comment:
After trying to install the module I see why you added the `base.group_multi_company` to the fields, but it seems to me like it doesn't fix the problem, the module still doesn't install due to similar error that we can see in the current red tests

I don't think these fields should have this group restriction since they should show up without multi-company, but from what I can tell its the 'check_company=True` on the original `product.supplierinfo` field that is causing it to inject itself with the `company_id` domain and erroring for `base.group_multi_company` permissions.

I can't figure out why this happens with this new model and not on the regular `product.supplierinfo`, `check_company=False` "fixes" the error but there has to be a better solution that doesn't fail since it makes no sense

Reply:
@pedrobaeza 
Can you help about this error, please?

Field 'product_tmpl_id' used in domain of <field name="product_id"> ([('product_tmpl_id', '=', product_tmpl_id)]) is restricted to the group(s) base.group_multi_company.

Reply:
Why an invisible field has domain? Try adding another invisible field without groups.

Reply:
It is tricky and I'm not sure how to cleanly get around it.
The domain gets added here https://github.com/odoo/odoo/blob/16.0/addons/product/models/product_supplierinfo.py#L21-L27 and due to the `check_company` on the field it errors.
This doesn't happen for supplierinfo so there has to be some way around it, unless it's a weird error case with inherited models, but I'd find that weird